General improvements

cppxor2arr opened the pull request.

cppxor2arr committed:
general improvements

schneiderchristopher commented:
Hello, thanks for this. I will check with #3 to proper handle the conflicts.

I guess the way I was doing the conditionals with match were wrong? Can you elaborate these changes? Thanks!

cppxor2arr committed:
do division to get correct value for unit

cppxor2arr commented on Cargo.toml:
```diff
@@ -1,6 +1,7 @@
 [package]
 name = "lsr"
 version = "0.1.0"
+edition = "2021"
```
Use latest edition (without this line `extern crate` declarations are needed to compile).

cppxor2arr commented:
Hi, this PR will also handle file sizes `1000`, `1000000`, etc., like PR #3. After seeing PR #3, I added a commit to this PR which also divides by the unit sizes to show the correct value for the unit.

Here's what I changed:

- Added `edition = "2021"` (latest edition) in **Cargo.toml** (without this, `extern crate` declarations are needed or else compile error)
- Removed unnecessary `extern crate` declarations
- In **main.rs** changed asterisk glob import to explicitly importing what's needed since that's preferred
- In **main.rs** changed `fs::{self}` to just `fs` in use declaration
- In **main.rs** changed `content.unwrap()` to `content?`
- In **paths.rs** in **`print()`** changed `match ... { true => {...} false => {...} }` to just `if ... {...} else {...}`
- In **paths.rs** in **`size_string_formatter()`** changed the `match` to just `if`, `else if`, `else` and removed redundant conditions (later I also made it show the correct value for each unit)
- In **paths.rs** in **`indentate_paths()`** changed `while` loop with mutable variable to `for` loop with range
